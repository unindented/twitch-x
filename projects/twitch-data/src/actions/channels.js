import {getAction} from '../api'

export function loadSearchChannels ({query, limit}) {
  const action = {
    type: 'LOAD_SEARCH_CHANNELS',
    payload: query
  }
  return getAction(action, '/search/channels', {query: {query: encodeURIComponent(query), limit}})
}
