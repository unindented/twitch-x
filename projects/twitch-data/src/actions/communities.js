import {getAction} from '../api'

export function loadTopCommunities ({limit}) {
  const action = {
    type: 'LOAD_TOP_COMMUNITIES'
  }
  return getAction(action, '/communities/top', {query: {limit}})
}
