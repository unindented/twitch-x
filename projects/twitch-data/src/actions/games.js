import {getAction} from '../api'

export function loadTopGames ({limit}) {
  const action = {
    type: 'LOAD_TOP_GAMES'
  }
  return getAction(action, '/games/top', {query: {limit}})
}

export function loadSearchGames ({query, limit}) {
  const action = {
    type: 'LOAD_SEARCH_GAMES',
    payload: query
  }
  return getAction(action, '/search/games', {query: {query: encodeURIComponent(query), limit}})
}
