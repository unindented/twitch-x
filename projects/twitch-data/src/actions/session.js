export function login (token) {
  return {
    type: 'LOGIN',
    payload: token
  }
}
