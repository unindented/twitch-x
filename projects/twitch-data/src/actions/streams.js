import {getAction} from '../api'

export function loadTopStreams ({limit}) {
  const action = {
    type: 'LOAD_TOP_STREAMS'
  }
  return getAction(action, '/streams', {query: {limit}})
}
