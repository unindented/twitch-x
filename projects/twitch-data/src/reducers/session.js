const initialState = {token: null}

export default (state = initialState, action = {}) => {
  switch (action.type) {
    case 'LOGIN':
      return {token: action.payload}
  }

  return state
}
