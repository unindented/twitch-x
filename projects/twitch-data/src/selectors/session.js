import {createSelector} from 'reselect'

export const getToken = createSelector(
  (state) => state.data.session.token,
  (token) => token
)
