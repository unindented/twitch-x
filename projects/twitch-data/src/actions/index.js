import * as session from './session'
export {session as sessionActions}
