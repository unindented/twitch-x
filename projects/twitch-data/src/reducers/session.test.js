import {login} from '../actions/session'
import reducer from './session'

describe('reducers/session', () => {
  describe('default', () => {
    it('returns initial state', () => {
      const prevState = undefined
      expect(reducer(prevState)).toMatchSnapshot()
    })
  })

  describe('LOGIN', () => {
    it('stores the token', () => {
      const prevState = {}
      expect(reducer(prevState, login('foo'))).toMatchSnapshot()
    })
  })
})
