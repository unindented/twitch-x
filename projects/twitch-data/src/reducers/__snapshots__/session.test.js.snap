// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/session LOGIN stores the token 1`] = `
Object {
  "token": "foo",
}
`;

exports[`reducers/session default returns initial state 1`] = `
Object {
  "token": null,
}
`;
